File: app/src/main/java/com/glsebastiany/bakingapp/view/recyclerview/RVListBaseAdapter.java
package com.glsebastiany.bakingapp.view.recyclerview;

import android.content.Context;

import java.util.List;

public abstract class RVListBaseAdapter<T> extends RVBaseAdapter {

    private List<T> list;

    @Override
    public Object getModelObjForPosition(Context context, int position) {
        if (list == null){
            return null;
        } else {
            return list.get(position);
        }
    }

    @Override
    public int getItemCount() {
        if (list == null){
            return 0;
        } else {
            return list.size();
        }
    }

    public void updateItems(List<T> items) {
        list = items;

        notifyDataSetChanged();
    }

    public int indexOf(T item){
        if (list == null) {
            return -1;
        } else {
            return list.indexOf(item);
        }
    }

}

File: app/src/main/java/com/glsebastiany/bakingapp/view/recyclerview/RVListBaseAdapter.kt
package com.glsebastiany.bakingapp.view.recyclerview

import android.content.Context

abstract class RVListBaseAdapter<T> : RVBaseAdapter() {

    private var list : List<T>? = listOf()

    override fun getModelObjForPosition(context: Context, position: Int): Any =
            list?.get(position) as Any

    override fun getItemCount(): Int = list?.count() ?: 0

    fun updateItems(items: List<T>?) {
        list = items

        notifyDataSetChanged()
    }

    fun indexOf(item: T): Int = list?.indexOf(item) ?: -1

}